compiler/src/iree/compiler/Modules/IO/Parameters/IR/IOParametersBase.td

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_DIALECT_MODULES_IO_PARAMETERS_BASE
+#define IREE_DIALECT_MODULES_IO_PARAMETERS_BASE
+
+include "iree/compiler/Dialect/Util/IR/UtilBase.td"
+
+//===----------------------------------------------------------------------===//
+// External parameter resource management
+//===----------------------------------------------------------------------===//
+
+def IOParameters_Dialect : Dialect {
+  let name = "io_parameters";
+  let cppNamespace = "::mlir::iree_compiler::IREE::IO::Parameters";
+
+  let summary = [{
+    External parameter resource management APIs.
+  }];
+  let description = [{
+    Parameters are externalized storage for resources that are
+    asynchronously accessible and device-aware. Parameters can be read or
+    written on the same device timelines as the operations that consume or
+    produce them and with locality pinning to ensure memory doesn't need to
+    move. Parameters are referenced by a scope and a key, with the scope
+    being optional but strongly recommended as a way to distinguish sets of
+    parameters that may exist when multiple model parts are compiled together
+    and would otherwise collide.
```

Comment:
I'd find an example or diagram useful to understand "when multiple model parts are compiled together and would otherwise collide" (a general example of "multiple model parts compiled together" would be nice too, independent of the comments here)